### ISSUE TITLE
when i train the second epoch on voc, location loss is going very high value and got a fualty loss.

### ISSUE THREAD

**Issue by QuestionLock** [gurkirt/FPN.pytorch1.0#5]
hi ,gurkirt:
    i benefit greatly from your work! but when i train the net on voc , the first epoch(about the first 1000  iterations）it works well. but when it was trained by the same data for the second time(the second epoch) the location loss gone faulty but the cls loss keep normal. however this cause the validation performance decreases greatly. i have tried many tricks dealing with loss problem and it dosen't works. it is strange that it is the problem happens at exactly the iteration belongs to the second epoch. could you help me? or do you kown where the problem lies. i use the lr0.0005 and default boxes, shared head,mbox loss. looking for your reply,thanks!

**Comment by gurkirt** [comment]
It is most likelly the high learning rate, 0.0005 is kind of on the edge, try 0.0003 or 0.0004.

**Comment by gurkirt** [comment]
@QuestionLock  did that solve your problem?

**Comment by QuestionLock** [comment]
> @QuestionLock did that solve your problem?

a lot thanks for your caring. i'm sorry that i forgot to tell you the solution. after contacting with you, i did a lot study . i found the problem was when training  the second epoch, the data in the dataset changed Unknowingly. i solve the problem by defining the detaset every epoch. it was a bit time consuming, but also worked.

**Comment by gurkirt** [comment]
I think the issue is that data in annot_info being modified boxes at https://github.com/gurkirt/FPN.pytorch1.0/blob/master/data/detectionDatasets.py#L96 is being updated in self.transform. Maybe use .copy on line 96 and 97.
